tests/integration/messages/generated/rflx-universal-message.adb

```diff
@@ -155,7 +155,9 @@ is
      (Sufficient_Buffer_Length (Ctx, Fld)
       and then (case Fld is
                    when F_Data | F_Option_Types | F_Options | F_Values =>
-                      Ctx.Buffer.all (RFLX_Types.To_Index (Field_First (Ctx, Fld)) .. RFLX_Types.To_Index (Field_Last (Ctx, Fld))) = Data,
+                      Data'Length = RFLX_Types.To_Index (Field_Last (Ctx, Fld)) - RFLX_Types.To_Index (Field_First (Ctx, Fld)) + 1
+                      and then (for all I in RFLX_Types.Index range RFLX_Types.To_Index (Field_First (Ctx, Fld)) .. RFLX_Types.To_Index (Field_Last (Ctx, Fld)) =>
+                                   Ctx.Buffer.all (I) = Data (Data'First + (I - RFLX_Types.To_Index (Field_First (Ctx, Fld))))),
```

Comment:
Comparing the length and the equality of each element of the array makes it easier to prove the code? I would never have thought of that!

Reply:
The definition of equality between arrays is expressed in the same way (equality of length + comparison of elements), but the slices add an extra layer of complexity (adds another quantifier inside). So by doing the comparison ourselves we can remove one layer.

Reply:
Interesting. Would it be possible that SPARK detects itself that this extra layer of complexity is not needed in this specific case and that it can do this transformation itself?

Reply:
Possibly, I created a SPARK ticket for that.

Reply:
Just to clarify, I don't think we should wait for such a SPARK enhancement. It will take time.

Reply:
I agree with you. How about adding a comment to the code generator? At some point, someone could wonder why checking the equality was solved this way.

Reply:
done